utils/certs/gen_certificates.sh

```diff
@@ -113,30 +116,35 @@ function generate_server_cert () {
 	openssl genrsa -out $CERTS/server.key 4096
 	chmod 0400 $CERTS/server.key
 	# Generate a Certificate Signing Request (CRS)
-	openssl req -new -config server.cnf -key $CERTS/server.key \
+	openssl req -new -config "$CONFIGS/server.cnf" -key $CERTS/server.key \
 		-out server.csr -batch
 	# Create Certificate from request
-	openssl ca -config ca.cnf -keyfile $PRIVATE/daosCA.key \
+	openssl ca -config "$CONFIGS/ca.cnf" -keyfile $PRIVATE/daosCA.key \
 		-cert $CERTS/daosCA.crt -policy signing_policy \
 		-extensions signing_server -out $CERTS/server.crt \
 		-outdir $CERTS -in server.csr -batch
 
-	echo "Required Shell Certificate Files:
+	echo "Required Server Certificate Files:
 	$CERTS/daosCA.crt
 	$CERTS/server.key
 	$CERTS/server.crt"
 }
 
 function generate_test_cert () {
+	# Don't try to generate the test cert in production installs.
+	if ! [ -f "$CONFIGS/test.cnf" ]; then
+		return
+	fi
+
```

Comment:
That works, but I think it would be more clear if the check was done in main() prior to calling generate_test_cert().  I wouldn't hold it up for this, but if there are any other changes you decide to make, then consider this one.

Reply:
If we did that, then we'd have to move knowledge about test certs into main, whereas here it's nicely self-contained.